### PR TITLE
bridge-flexibility

### PR DESCRIPTION
**Pull request by sloops77**
Purpose is to formalise use of headers on the session and allow API access to both 
* the headers that should be added automatically on each request for the session
* which urls the headers should be added to

### CODE REVIEW

File: packages/core/src/utils/match.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+import some from 'lodash/some';
+
+export default function match(url, patternOrArray) {
+  if (patternOrArray === undefined) {
     return true;
   }
 
-  return url.match(pattern);
+  const patternsArray = isArray(patternOrArray) ? patternOrArray : [patternOrArray];
+  return some(
```
Comment (YosBD) [inline packages/core/src/utils/match.js]: [minor] you can do it with one iteration:
```
return some(patternsArray, pattern => url.match(isString(pattern) ? new RegExp(`^${pattern}$`) : pattern))
```

File: packages/core/src/utils/match.js [hunk 2]
```diff
@@ -1,7 +1,17 @@
-export default function match(url, pattern) {
-  if (pattern === undefined) {
+import isArray from 'lodash/isArray';
+import isString from 'lodash/isString';
+import map from 'lodash/map';
+import some from 'lodash/some';
+
+export default function match(url, patternOrArray) {
+  if (patternOrArray === undefined) {
```
Comment (YosBD) [inline packages/core/src/utils/match.js]: maybe the pattern to `always match` should be asterisk?

File: packages/server-bridge/src/bridge.js [hunk 3]
```diff
@@ -63,6 +65,9 @@ function startSession(log, source, socket) {
       log.info(`${session.logPrefix}: ended session`);
       delete sessions[socketId];
     }],
+    setSessionHeaders(headers) {
+      this.headers = assign({}, headers, { bridges: [source] });
```
Comment (YosBD) [inline packages/server-bridge/src/bridge.js]: Nice!
If I understand correctly, these headers will be assigned to each request headers, right?

Comment (YosBD) [reply packages/server-bridge/src/bridge.js]: And as it is only supports a single bridge, maybe it's better to call it `bridge` and to not support arrays

File: packages/core/src/utils/match.js (hunk 2 above)
Comment (sloops77) [reply packages/core/src/utils/match.js]: look at the tests - the api for adding a url that applies to all urls is super obvious
`request.use((req, res, next) => next())`

File: packages/server-bridge/src/bridge.js (hunk 3 above)
Comment (sloops77) [reply packages/server-bridge/src/bridge.js]: yes & ok

**Pull request merged**